src/runtime/inline-cta-api.ts

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright 2025 The Subscribe with Google Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ActionToIframeMapping, parseUrl} from '../utils/url';
+import {ActivityIframeView} from '../ui/activity-iframe-view';
+import {ActivityPorts} from '../components/activities';
+import {ClientConfigManager} from './client-config-manager';
+import {Deps} from './deps';
+import {Doc} from '../model/doc';
+import {EntitlementsManager} from './entitlements-manager';
+import {Intervention} from './intervention';
+import {ProductType} from '../api/subscriptions';
+import {assert} from '../utils/log';
+import {feArgs, feUrl} from './services';
+import {setImportantStyles} from '../utils/style';
+
+const INLINE_CTA_ATTRIUBUTE_QUERY = 'div[rrm-inline-cta]';
+const INLINE_CTA_ATTRIUBUTE = 'rrm-inline-cta';
+const DEFAULT_PRODUCT_TYPE = ProductType.UI_CONTRIBUTION;
+
+export class InlincCtaApi {
+  private readonly doc_: Doc;
+  private readonly win_: Window;
+  private readonly activityPorts_: ActivityPorts;
+  private readonly clientConfigManager_: ClientConfigManager;
+  private readonly entitlementsManager_: EntitlementsManager;
+  constructor(private readonly deps_: Deps) {
+    this.doc_ = deps_.doc();
+    this.win_ = deps_.win();
+    this.activityPorts_ = deps_.activities();
+    this.entitlementsManager_ = deps_.entitlementsManager();
+    this.clientConfigManager_ = deps_.clientConfigManager();
+    assert(
+      this.clientConfigManager_,
+      'InlineCta requires an instance of ClientConfigManager.'
+    );
+  }
+
+  private actionToUrlPrefix_(
+    configId: string,
+    actions: Intervention[]
+  ): string {
+    for (const action of actions) {
```

Comment:
nit: maybe somethign like 

```
const action = actions.find((action) => action.configurationId === configId));
return ActionToIframeMapping[action?.type] ?? '';
```

Reply:
Will do it in follow up cl! My git it acting weird and cannot push my changes.